src/lib/createStardustComponent.tsx

```diff
@@ -0,0 +1,36 @@
+import createComponentInternal from './createComponent'
+import * as React from 'react'
+import * as _ from 'lodash'
+import { ComponentSlotClasses } from '../themes/types'
+
+export interface RenderStardustResultConfig {
+  classes: ComponentSlotClasses
+  rtl: boolean
+}
+
+export type RenderStardustComponentCallback<P> = (
```

Comment:
I would just inline this type to the render method, it's lightweight and it'd make things more readable